bot/receiver.go

```diff
@@ -30,82 +31,94 @@ func playAudioCmd(ctx CommandCtx) error {
 			log.Printf("Error joining VC to play audio: %s", err)
 			return ctx.Reply("Failed to join voice channel.")
 		}
-
+		ar := NewAudioReceiver(voiceConn)
+		activeReceiver = ar
 		go monitorVoiceActivity(s, gId, vc)
-		go StartAudioReceiver(voiceConn)
+		go ar.Start()
 		return ctx.Reply("Venova has entered the chat...")
 	}
 	return nil
 }
 
+func NewAudioReceiver(vc *discordgo.VoiceConnection) *AudioReceiver {
```

Comment:
You don't have to do this, but it's something to consider: What if most of `ar.Start()` happened in this function, and we returned a type of `(*AudioReceiver, error)` instead? Then instead of `Start()` we might have a `Run()` function which would just be the big `for { ...  }` loop from the current version of `Start()`. This reduces a lot of the error handling we're doing in a different goroutine (where errors are usually much harder to work with and bubble up to our main thread/routine) and simplifies the overall behavior.